src/realm/group.cpp

```diff
@@ -120,48 +123,34 @@ void Group::upgrade_file_format(int target_file_format_version)
     // Be sure to revisit the following upgrade logic when a new file foprmat
     // version is introduced. The following assert attempt to help you not
     // forget it.
-    REALM_ASSERT_EX(target_file_format_version == 7, target_file_format_version);
+    REALM_ASSERT_EX(target_file_format_version == s_current_file_format_version, target_file_format_version);
```

Comment:
As we just spoke about, this change defies the original intend of the assert: to remember to check the following code for _every_ file format bump.